introducing-pgzx.mdx

```diff
@@ -0,0 +1,273 @@
+
+Another advantage of memory contexts is that they improve [memory monitoring](https://www.postgresql.org/docs/current/view-pg-backend-memory-contexts.html), because context have names and you can see how much memory is used by each context. This is useful for debugging large memory usage.
+
+This model of using an arena/context allocator happens to match really good with Zig. One reason is Zig's convention to have all functions/objects that allocate memory receive an allocator as a parameter. This makes allocations [more explicit](https://notes.eatonphil.com/2024-03-15-zig-rust-and-other-languages.html) but also makes it easy to use a custom allocator. `pgzx` defines custom allocators that wrap the Postgres memory contexts and make them available to Zig code.
+
+Here is an example that creates a new context as a child of the current context and gets the allocator for it:
+
+```zig
+var memctx = try pgzx.mem.createAllocSetContext("zig_context", .{ .parent = pg.CurrentMemoryContext });
+const allocator = memctx.allocator();
+```
+
+### Error handling
+
+Another Postgres API that you will very likely need to know about in a more complex extension is the error handling. Postgres implements "exceptions" in C via `setjmp/longjmp` and provides a set of macros to throw and catch them ([PG_TRY/PG_CATCH](https://github.com/postgres/postgres/blob/master/src/include/utils/elog.h#L318)).
+
+The issue is that long jumps could bypass the Zig control flow, for example `errdefer` blocks might not be executed. This means that if your extension calls into Postgres APIs, and those APIs can throw errors, long jumps might skip your `defer` and `errdefer` blocks!
+
+Luckily, pgzx is here to help. It provides a set of functions that allow you to catch the Postgres exceptions and convert them into Zig errors. For example:
+
+```zig
+var errctx = pgzx.err.Context.init();
+defer errctx.deinit();
+if (errctx.pg_try()) {
+    // zig code that calls several Postgres C functions.
+} else {
+    return errctx.errorValue();
+}
+```
+
+### Development Environment
+
+pgzx comes with a [Nix flakes](https://nixos.wiki/wiki/Flakes) based development environment for developing extensions as well as `pgzx` itself. It also comes with a project template which you can use to set up this environment in a new repository. To use it, [install Nix](https://github.com/DeterminateSystems/nix-installer) and then run:
```

Comment:
Do we want to point users to the docker container we provide as well? Via `./dev/docker/build.sh` you can build a container with the nix development shell. So you can test development with pgzx without having to install nix.